CMakeLists.txt

```diff
@@ -408,6 +408,10 @@ include (CheckIncludeFileCXX)
 check_include_file_cxx (execinfo.h SEASTAR_EXECINFO)
 check_include_file_cxx (sys/sdt.h SEASTAR_SDT_H)
 configure_file (config.hh.in ${CMAKE_CURRENT_SOURCE_DIR}/include/seastar/config.hh @ONLY)
+pkg_check_modules (UCONTEXT IMPORTED_TARGET libucontext)
+if (UCONTEXT_FOUND)
+  set (UCONTEXT_LIB PkgConfig::UCONTEXT)
+endif ()
```

Comment:
i'd extract this into a separate `Finducontext.cmake`. it would resemble `Findrt.cmake`.

Reply:
okay will have a look

Reply:
I will push with the changes in new PR removed (I rebased it onto this branch)

Reply:
also, the introduction of `config.hh` is an important design decision. currently, we are using macros to persist various settings, and to pass them down to the Seastar applications. i think both of them have pros and cons. personally, i am in favor of `config.hh`. but i think we need more inputs on this regard. as this would be a tree-wide change if we want to be consistent on persisting / populating the configurations.

Reply:
okay, I wasn’t sure how to pass the defines down into the source code from cmake